app/src/main/kotlin/github/com/rhacco/dota2androidapp/activities/LiveMatchesActivity.kt

```diff
@@ -29,21 +39,26 @@ class LiveMatchesActivity : BaseLifecycleActivity<MatchesViewModel>() {
     }
 
     // Add tournament matches first, then sort by average MMR (descending)
-    private fun updateLiveMatchesListData(listItemData: LiveMatchesListAdapter.ListItemData) {
-        if (listItemData.mAverageMMR < 1) {
-            mListAdapter.mListItemsData.add(0, listItemData)
+    private fun updateLiveMatchesListData(newItem: LiveMatchesListAdapter.ListItemData) {
+        // Strangely, sometimes the same match arrives here twice, so prevent creating duplicates
+        mListAdapter.mListItemsData
+                .filter { it.mMatchID == newItem.mMatchID }
+                .forEach { return }
+
+        if (newItem.mAverageMMR < 1) {
```

Comment:
Why is this mAverageMMR evaluation needed?

Reply:
Tournament matches are also returned by GetTopLiveGame and they will have an average MMR of 0. So this evaluation checks if it needs to handle a tournament or a ranked match (which will have average MMR > 0)

Reply:
Please comment this observation in the code as this detail is not self-evident in the logic. In the future, we maybe put this in a helper.

Reply:
I agree, thanks. Done in #9